src/hooks/useSearchQuery.jsx

```diff
@@ -0,0 +1,29 @@
+import { useState, useEffect, useCallback } from 'react';
+import { useNavigate, useSearchParams } from 'react-router-dom';
+
+export const useSearchQuery = () => {
+  const [query, setQuery] = useState('');
+  const navigate = useNavigate();
+  const [searchParams] = useSearchParams();
+
+  useEffect(() => {
+    const queryParam = searchParams.get('query') || '';
+    setQuery(queryParam);
+  }, [searchParams]);
+
+  const updateQuery = useCallback(
```

Comment:
to update search param you should use the setSearchParams function returned from `useSearchParams()` hook.

https://reactrouter.com/6.28.1/hooks/use-search-params